qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -161,10 +165,12 @@ def __init__(self, backend, job_id, api, qobj=None,
             validate_qobj_against_schema(qobj)
             self._qobj_payload = qobj.to_dict()
             self._status = JobStatus.INITIALIZING
+            self._job_name = job_name
```

Comment:
Instead of setting an attribute, can you explore just passing the name to `.submit()`? As you mention, introducing a class-level attribute can be a bit confusing both for users ... and for maintainers (specially since the attribute is currently _not_ set for retrieved jobs, and the job class is quite convoluted). This hopefully will be enough for covering the requirements for this iterations, and we can include the name in a more controlled way during #104 .